### PR TITLE
feat: useStorageState setStoreValue catch this error to the developer

### LINKED ISSUE #2100: feat: useSessionStorageState throwing an exception with an error
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] #2131
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

1. When adding a value to sessionStorage via the custom useStorageState hook, you may encounter the error "Failed to execute 'setItem' on 'Storage': Setting the value of 'storedFile' exceeded the quota". The hook has a method for adding a value to SS(setStoredValue), where it can be seen that in case of an error, in the catch block, the error is simply output to console.log. I would like to be able to catch this error to the developer.
2. Screenshot 2023-03-07 at 09 42 10
3.  My suggestions:  make it possible to set an errorCallback to which to pass an error, or throw an exception
-->

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  useStorageState catch SessionStorage error  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### PR DESCRIPTION
**Pull request by wangping51**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close: #2100 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   The setStoreValue error of useLocalStorageState and useSessionStorageState can be captured by the developer using try/catch     |
| 🇨🇳 Chinese |  useLocalStorageState 和 useSessionStorageState 的 setStoreValue 错误开发者可以使用 try/catch 进行捕获  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by liuyib** [comment]
This not a better way to resolve #2100. You known, we can't try...catch a hook in React, so I suggest to use a callback (e.g. `onError`, refer: [#L107](https://github.com/vueuse/vueuse/blob/7b27148433df84a8e791682d955a8d37a5105023/packages/core/useStorage/index.ts#L107)) @51wangping

**Comment by wangping51** [comment]
> This not a better way to resolve #2100. You known, we can't try...catch a hook in React, so I suggest to use a callback (e.g. `onError`, refer: [#L107](https://github.com/vueuse/vueuse/blob/7b27148433df84a8e791682d955a8d37a5105023/packages/core/useStorage/index.ts#L107)) @51wangping

Changed

File: packages/hooks/src/useLocalStorageState/index.en-US.md [hunk 1]
```diff
@@ -40,11 +40,12 @@ const [state, setState] = useLocalStorageState<T>(
...
-| ------------ | ----------------------------- | ------------------------ | ---------------- |
-| defaultValue | Default value                 | `any \| (() => any)`     | -                |
-| serializer   | Custom serialization method   | `(value: any) => string` | `JSON.stringify` |
-| deserializer | Custom deserialization method | `(value: string) => any` | `JSON.parse`     |
+| Property     | Description                   | Type                       | Default                    |
+| ------------ | ----------------------------- | -------------------------- | -------------------------- |
+| defaultValue | Default value                 | `any \| (() => any)`       | -                          |
+| serializer   | Custom serialization method   | `(value: any) => string`   | `JSON.stringify`           |
+| deserializer | Custom deserialization method | `(value: string) => any`   | `JSON.parse`               |
+| onError      | On error callback             | `(error: unknown) => void` | `(e) => {console.error(e)` |
```
Comment (liuyib) [inline packages/hooks/src/useLocalStorageState/index.en-US.md:49]: the default value `(e) => {console.error(e)` has a syntax error, should be: `(e) => { console.error(e) }`, right?

File: packages/hooks/src/useLocalStorageState/index.zh-CN.md [hunk 2]
```diff
@@ -40,11 +40,12 @@ const [state, setState] = useLocalStorageState<T>(
...
-| ------------ | ------------------ | ------------------------ | ---------------- |
-| defaultValue | 默认值             | `any \| (() => any)`     | -                |
-| serializer   | 自定义序列化方法   | `(value: any) => string` | `JSON.stringify` |
-| deserializer | 自定义反序列化方法 | `(value: string) => any` | `JSON.parse`     |
+| 参数         | 说明               | 类型                       | 默认值                     |
+| ------------ | ------------------ | -------------------------- | -------------------------- |
+| defaultValue | 默认值             | `any \| (() => any)`       | -                          |
+| serializer   | 自定义序列化方法   | `(value: any) => string`   | `JSON.stringify`           |
+| deserializer | 自定义反序列化方法 | `(value: string) => any`   | `JSON.parse`               |
+| onError      | 错误回调函数       | `(error: unknown) => void` | `(e) => {console.error(e)` |
```
Comment (liuyib) [inline packages/hooks/src/useLocalStorageState/index.zh-CN.md:49]: same as above

File: packages/hooks/src/useLocalStorageState/index.en-US.md (hunk 1 above)
Comment (wangping51) [reply packages/hooks/src/useLocalStorageState/index.en-US.md:49]: I'm sorry

Comment (wangping51) [reply packages/hooks/src/useLocalStorageState/index.en-US.md:49]: Modified

**Comment by crazylxr** [comment]
冲突解决一下，代码应该没啥问题了

**Comment by crazylxr** [comment]
冲突解决一下，代码应该没啥问题了

**Review by liuyib — APPROVED** [review]

**Comment by liuyib** [comment]
> 冲突解决一下，代码应该没啥问题了

done

**Pull request merged**